### PR TITLE
Move `test-snaps` site to this repository

### LINKED ISSUE #1546: Move `test-snaps` site to this repository

### PR DESCRIPTION
**Pull request by Mrtenz**
This moves the `test-snaps` site to this repository. It's largely the same as the `MetaMask/test-snaps` site, with some small differences:

- Some copy was updated to make more sense. For example: "Send Test to Snap" was changed to "Get Public Key".
- All snap IDs and ports were updated to work with the new examples.
- Some breaking changes were handled in changes to the new examples.
- The Flask fox icon was changed to the regular fox (recommended by @eriknson).
- SCSS was removed in favour of CSS. We were not using any SCSS features.
- Some files were moved around.
   - All snaps are moved to `features/snaps`.
   - Within each snap folder is now a `constants.ts` file containing the snap ID and port.
   - Where applicable, snap-specific components were moved to a `components` folder inside the snap folder.

Closes #1546.

### CODE REVIEW

File: .github/workflows/publish-github-pages.yml [hunk 1]
```diff
@@ -0,0 +1,49 @@
+name: Publish execution environment
```
Comment (FrederikBolding) [inline .github/workflows/publish-github-pages.yml:1]: This seems wrong?

File: packages/test-snaps/LICENSE [hunk 2]
```diff
@@ -0,0 +1,15 @@
+ISC License
```
Comment (FrederikBolding) [inline packages/test-snaps/LICENSE:1]: Should this be ISC licensed?

File: yarn.lock [hunk 3]
```diff
@@ -4538,8 +4538,8 @@ __metadata:
   linkType: hard
 
 "@metamask/providers@npm:^11.0.0":
-  version: 11.1.0
-  resolution: "@metamask/providers@npm:11.1.0"
+  version: 11.0.0
```
Comment (FrederikBolding) [inline yarn.lock:4541]: Why the revert 🤔

File: .yarnrc.yml [hunk 4]
```diff
@@ -22,4 +22,4 @@ yarnPath: .yarn/releases/yarn-3.4.1.cjs
 packageExtensions:
   '@wdio/browser-runner@*':
     dependencies:
-      '@babel/core': '*'
+      '@babel/core': 'patch:@babel/core@npm%3A7.21.0#./.yarn/patches/@babel-core-npm-7.21.0-fb3817b0e5.patch::locator=root%40workspace%3A'
```
Comment (FrederikBolding) [inline .yarnrc.yml:25]: ?

File: yarn.lock (hunk 3 above)
Comment (Mrtenz) [reply yarn.lock:4541]: Not intentional, not sure why this happened.

File: .yarnrc.yml (hunk 4 above)
Comment (Mrtenz) [reply .yarnrc.yml:25]: I added some more stuff to `packageExtensions` and replaced the Babel version with the one we use in `package.json`, but later ended up removing those additions to `packageExtensions`. 😅

File: packages/test-snaps/LICENSE (hunk 2 above)
Comment (Mrtenz) [reply packages/test-snaps/LICENSE:1]: Probably? It's not really intended to be used as example.

Comment (FrederikBolding) [reply packages/test-snaps/LICENSE:1]: Test dapp is MIT licensed, perhaps we should follow that? 🤔

**Review by FrederikBolding — APPROVED** [review]

**Comment by Mrtenz** [comment]
@SocketSecurity ignore-all

**Pull request merged**